storey/sources.py

```diff
@@ -903,3 +903,111 @@ def _init(self):
             else:
                 df = pandas.read_parquet(path, columns=self._columns, storage_options=self._storage_options)
             self._dfs.append(df)
+
+
+class SQLSource(_IterableSource, WithUUID):
+    """Use dql table as input source for a flow.
+
+
+    :parameter key_field: the primary key of the table.
+    :parameter time_field: column to be used as time for events.
+    :parameter id_field: column to be used as ID for events.
+    :parameter db_name: url string connection to sql database.
+    :parameter table_name: the name of the table to access,
+                                from the current database
+    :parameter start_filter: If not None, the results will be filtered by partitions and 'filter_column' > start_filter.
+                            Default is None (not in use)
+    :parameter end_filter:  If not None, the results will be filtered by partitions 'filter_column' <= end_filter.
+                            Default is None (not in use)
+
+    """
+
+    def __init__(
+        self,
+        db_path: str = None,
+        table_name: str = None,
+        key_field: Optional[Union[str, List[str]]] = None,
+        start_filter: Optional[datetime] = None,
+        end_filter: Optional[datetime] = None,
+        time_field: Optional[str] = None,
+        id_field: Optional[str] = None,
+        **kwargs,
+    ):
+
+        if key_field is not None:
+            kwargs["key_field"] = key_field
+        if time_field is not None:
+            kwargs["time_field"] = time_field
+        if id_field is not None:
+            kwargs["id_field"] = id_field
+        _IterableSource.__init__(self, **kwargs)
+        WithUUID.__init__(self)
+
+        if not all([db_path, table_name]):
+            raise ValueError("cannot specify without db_path and table_name args")
+
+        self.table_name = table_name
+        self.db_path = db_path
+
+        self._key_field = key_field
+        if isinstance(time_field, str) and "." in time_field:
+            self._time_field = time_field.split(".")
+        else:
+            self._time_field = time_field
+        if isinstance(id_field, str) and "." in id_field:
+            self._id_field = id_field.split(".")
+        else:
+            self._id_field = id_field
+
+    async def _run_loop(self):
+        import sqlalchemy as db
+
+        engine = db.create_engine(self.db_path)
+        metadata = db.MetaData()
+        connection = engine.connect()
+        table = db.Table(self.table_name, metadata, autoload=True, autoload_with=engine)
+        results = connection.execute(db.select([table])).fetchall()
+        df = pandas.DataFrame(results)
```

Comment:
1. Why not use the results directly? Why do we need an intermediate dataframe?
2. If we did need it, we shouldn't duplicate all the code for ingesting a dataframe (below).